Core/include/Acts/TrackFitting/BetheHeitlerApprox.hpp

```diff
@@ -0,0 +1,322 @@
+      for (int i = 0; i < NComponents; ++i) {
+        // These transformations must be applied to the data according to ATHENA
+        // (TrkGaussianSumFilter/src/GsfCombinedMaterialEffects.cxx:79)
+        if (transform) {
+          ret[i] = detail::inverseTransformComponent(
+              poly(xx, data[i].weightCoeffs), poly(xx, data[i].meanCoeffs),
+              poly(xx, data[i].varCoeffs));
+        } else {
+          ret[i].weight = poly(xx, data[i].weightCoeffs);
+          ret[i].mean = poly(xx, data[i].meanCoeffs);
+          ret[i].var = poly(xx, data[i].varCoeffs);
+        }
+
+        weight_sum += ret[i].weight;
+      }
+
+      for (int i = 0; i < NComponents; ++i) {
+        ret[i].weight /= weight_sum;
+      }
+
+      return ret;
+    };
+
+    // Return no change
+    if (x < noChangeLimit) {
+      std::array<detail::GaussianComponent, NComponents> ret{};
+
+      ret[0].weight = 1.0;
+      ret[0].mean = 1.0;  // p_initial = p_final
+      ret[0].var = 0.0;
+
+      return ret;
+    }
+    // Return single gaussian approximation
+    if (x < singleGaussianLimit) {
+      std::array<detail::GaussianComponent, NComponents> ret{};
+      ret[0] = BetheHeitlerApproxSingleCmp::mixture(x)[0];
+      return ret;
+    }
+    // Return a component representation for lower x0
+    if (x < lowerLimit) {
+      return make_mixture(m_low_data, x, m_low_transform);
+    }
+    // Return a component representation for higher x0
+    else {
+      // Cap the x because beyond the parameterization goes wild
+      const auto high_x = std::min(higherLimit, x);
+      return make_mixture(m_high_data, high_x, m_high_transform);
+    }
+  }
+
+  /// Loads a parameterization from a file according to the Atlas file
+  /// description
+  ///
+  /// @param low_parameters_path Path to the foo.par file that stores
+  /// the parameterization for low x/x0
+  /// @param high_parameters_path Path to the foo.par file that stores
+  /// the parameterization for high x/x0
+  static auto loadFromFile(const std::string &low_parameters_path,
+                           const std::string &high_parameters_path) {
+    auto read_file = [](const std::string &filepath) {
+      std::ifstream file(filepath);
+
+      if (!file) {
+        throw std::invalid_argument("Could not open '" + filepath + "'");
+      }
+
+      std::size_t n_cmps, degree;
+      bool transform_code;
+
+      file >> n_cmps >> degree >> transform_code;
+
+      if (NComponents != n_cmps) {
+        throw std::invalid_argument("Wrong number of components in '" +
+                                    filepath + "'");
+      }
+
+      if (PolyDegree != degree) {
+        throw std::invalid_argument("Wrong wrong polynom order in '" +
+                                    filepath + "'");
+      }
+
+      if (!transform_code) {
+        throw std::invalid_argument("Transform-code is required in '" +
+                                    filepath + "'");
+      }
```

Comment:
does this mean we only support `transform_code = false`?

Reply:
yeah at the moment yes. The Atlas "file specification"  (which is the worst file specification I've ever seen) requires this, but there is not file present that has this...
But should be not very difficult to implement, I think about it.